src/char/char.cpp

```diff
@@ -1410,32 +1384,53 @@ int char_make_new_char_sql(struct char_session_data* sd, char* name_, int str, i
 	if( flag < 0 )
 		return flag;
 
-	//check other inputs
-#if PACKETVER >= 20120307
+	// Check inputs from the client - never trust it!
+
+	// Check the slot
+	if( slot < 0 || slot >= sd->char_slots ){
+		return -4; // invalid slot
+	}
+
+	// Check gender
 #if PACKETVER >= 20151001
```

Comment:
This packetver check needs to be removed, since now all clients use the switch to work.